dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -0,0 +1,287 @@
+package com.datadog.gradle.plugin
+
+import java.io.File
+import java.util.stream.Stream
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+
+internal class DdAndroidGradlePluginFunctionalTest {
+    @TempDir
+    lateinit var testProjectDir: File
+    lateinit var mainSrcDir: File
+    lateinit var settingsFile: File
+    lateinit var buildFile: File
+    lateinit var manifestFile: File
+    lateinit var gradlePropertiesFile: File
+
+    @BeforeEach
+    fun `set up`() {
+        mainSrcDir = File(testProjectDir, "src/main").apply { mkdirs() }
+        settingsFile = File(testProjectDir, "settings.gradle")
+        buildFile = File(testProjectDir, "build.gradle")
+        gradlePropertiesFile = File(testProjectDir, "gradle.properties")
+        manifestFile = File(mainSrcDir, "AndroidManifest.xml")
+        stubFile(settingsFile, SETTINGS_FILE_CONTENT)
+        stubFile(manifestFile, ANDROID_MANIFEST_FILE_CONTENT)
+        stubFile(gradlePropertiesFile, GRADLE_PROPERTIES_FILE_CONTENT)
+    }
+
+    // region Assemble
+
+    @Test
+    fun `M success W assemble`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_datadog_dep.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("assemble")
+            .withPluginClasspath()
```

Comment:
It seems there is no need to publish to the maven local if we use this method? If I get explanation here https://docs.gradle.org/current/javadoc/org/gradle/testkit/runner/GradleRunner.html#withPluginClasspath-- right